cmd/dra-example-kubeletplugin/state.go

```diff
@@ -173,3 +193,67 @@ func (s *DeviceState) prepareDevices(claim *resourceapi.ResourceClaim) (Prepared
 func (s *DeviceState) unprepareDevices(claimUID string, devices PreparedDevices) error {
 	return nil
 }
+
+func (pds PreparedDevices) GetDevices() []*drapbv1.Device {
+	var devices []*drapbv1.Device
+	for _, pd := range pds {
+		devices = append(devices, &pd.Device)
+	}
+	return devices
+}
+
+type PtrHasGroupVersionKind[T any] interface {
+	*T
+	GroupVersionKind() schema.GroupVersionKind
+}
+
+// GetMergedOpaqueDeviceConfig merges configs across device classes and resource claims for a given device request.
+func GetMergedOpaqueDeviceConfig[T any, TI PtrHasGroupVersionKind[T]](request string, claim *resourceapi.ResourceClaim, classes map[string]*resourceapi.DeviceClass) (T, error) {
+	// Collect all configs in reverse priority order. Device class configs have
+	// lower priority than resource claim configs. Within either a device class
+	// or a resource claim, configs earlier in the list have lower priority
+	// than configs later in the list.
+	var allConfigs []resourceapi.DeviceConfiguration
+	for _, config := range classes[request].Spec.Config {
+		allConfigs = append(allConfigs, config.DeviceConfiguration)
+	}
+	for _, config := range claim.Spec.Devices.Config {
+		if !slices.Contains(config.Requests, request) {
+			continue
+		}
+		allConfigs = append(allConfigs, config.DeviceConfiguration)
+	}
+
+	// Merge all configs from the allConfigs slice in the order they were
+	// added. Fields in a config that is unmarshaled earlier will be
+	// overwritten by fields in a config that is unmarhaled later.
+	var mergedConfig T
+	for _, config := range allConfigs {
+		if config.Opaque == nil {
+			continue
+		}
+		if config.Opaque.Driver != DriverName {
+			continue
+		}
```

Comment:
Should filtering be a separate operation to make it clearer that it's being applied? Especially since we're using a global variable for filtering here.

Reply:
I've made it a parameter so that the function is more generic.